components/inputs/input-number.js

```diff
@@ -81,32 +70,69 @@ class InputNumber extends LitElement {
 	}
 
 	render() {
-		const ariaRequired = this.required ? 'true' : undefined;
-
 		return html`
-			<label
-				class="${this.label && !this.labelHidden ? 'd2l-input-label' : 'd2l-offscreen'}"
-				for="${this._inputId}">${this.label}</label>
-			<input
-				aria-required="${ifDefined(ariaRequired)}"
+			<d2l-input-text
 				autocomplete="${ifDefined(this.autocomplete)}"
 				?autofocus="${this.autofocus}"
 				@change="${this._handleChange}"
-				class="d2l-input"
+				@d2l-form-element-should-validate="${this._handleNestedFormElementValidation}"
 				?disabled="${this.disabled}"
+				.forceInvalid="${this.invalid}"
 				id="${this._inputId}"
+				label="${this.label}"
+				?label-hidden="${this.labelHidden}"
 				name="${ifDefined(this.name)}"
 				placeholder="${ifDefined(this.placeholder)}"
-				type="text"
-				.value="${this._formattedValue}">
+				?required="${this.required}"
+				.value="${this._formattedValue}"
+			></d2l-input-text>
+			${ this.validationError ? html`<d2l-tooltip for=${this._inputId} state="error" align="start">${this.validationError}</d2l-tooltip>` : null }
 		`;
 	}
 
+	focus() {
+		this.shadowRoot.querySelector('d2l-input-text').focus();
+	}
+
+	async validate(showErrors) {
+		const errors = await super.validate(showErrors);
+		if (errors.length !== 0) {
+			return errors;
+		}
+		return await this.shadowRoot.querySelector('d2l-input-text').validate(showErrors);
+	}
```

Comment:
I lead you slightly astray here so apologies for that. It looks like the PR was slightly out of date so there are a couple changes that should be made here. [This is what the most up to date one looks like](https://github.com/BrightspaceUI/core/blob/master/components/inputs/input-date-range.js#L211).
1. We should call `this.shadowRoot.querySelector('d2l-input-text').validate(showErrors)` first since we want to ensure the `input-text` passes validation first before we perform number validation on it.
1. We always want to call `validate` on both since validation both shows new errors AND hides errors that are no longer relevant. But if validation fails for the child then we just want to suppress number validation errors with `ValidationType.SUPPRESS_ERRORS`
1. `showErrors` should be `validationType`.

I believe this will do what you need:
```
	async validate(validationType) {
		const childErrors = await this.shadowRoot.querySelector('d2l-input-text').validate(validationType);
		const errors = await super.validate(childErrors.length > 0 ? ValidationType.SUPPRESS_ERRORS : validationType);
		return [...childErrors, ...errors];
	}
```